src/clients/InsightsQueryClient.ts

```diff
@@ -0,0 +1,117 @@
+  }
+
+  // Block api
+  async getBlocks(req: Insights.QueryGetBlocksRequest): Promise<Insights.InsightsQueryResponse<Insights.QueryGetBlocksResponse>> {
+    const request = this.apiManager.path('block/list', {}, req)
+    const response = await request.get()
+    return response.data as Insights.InsightsQueryResponse<Insights.QueryGetBlocksResponse>
+  }
+
+  async getBlock(req: Insights.QueryGetBlockRequest): Promise<Insights.InsightsQueryResponse<Insights.QueryGetBlockResponse>> {
+    const request = this.apiManager.path('block/list', req)
+    const response = await request.get()
+    return response.data as Insights.InsightsQueryResponse<Insights.QueryGetBlockResponse>
+  }
+
+  async getBlockAtUnix(req: Insights.QueryGetBlockAtUnixRequest): Promise<Insights.InsightsQueryResponse<Insights.QueryGetBlockResponse>> {
+    const request = this.apiManager.path('block/unix', req)
+    const response = await request.get()
+    return response.data as Insights.InsightsQueryResponse<Insights.QueryGetBlockResponse>
+  }
+
+  async getBlockTime(): Promise<Insights.InsightsQueryResponse<Insights.QueryGetBlockTimeResponse>> {
+    const request = this.apiManager.path('block/time')
+    const response = await request.get()
+    return response.data as Insights.InsightsQueryResponse<Insights.QueryGetBlockTimeResponse>
+  }
+
+  // Transaction api
+  async getTransactions(req: Insights.QueryGetTransactionsRequest): Promise<Insights.InsightsQueryResponse<Insights.QueryGetTransactionsResponse>> {
+    const request = this.apiManager.path('tx/list', {}, req)
+    const response = await request.get()
+    return response.data as Insights.InsightsQueryResponse<Insights.QueryGetTransactionsResponse>
+  }
+
+  async getTransactionTypes(): Promise<Insights.InsightsQueryResponse<Insights.QueryGetTransactionTypesResponse>> {
+    const request = this.apiManager.path('tx/types')
+    const response = await request.get()
+    return response.data as Insights.InsightsQueryResponse<Insights.QueryGetTransactionTypesResponse>
+  }
+
+  async getTransactionActivity(req: Insights.QueryGetTransactionActivityRequest): Promise<Insights.InsightsQueryResponse<Insights.QueryGetTransactionActivityResponse>> {
+    const request = this.apiManager.path('tx/activity', {}, req)
+    const response = await request.get()
+    return response.data as Insights.InsightsQueryResponse<Insights.QueryGetTransactionActivityResponse>
+  }
+
+  // Balance api
+  async getTotalBalances(): Promise<Insights.InsightsQueryResponse<Insights.QueryGetTotalBalancesResponse>> {
+    const request = this.apiManager.path('balance/total')
+    const response = await request.get()
+    return response.data as Insights.InsightsQueryResponse<Insights.QueryGetTotalBalancesResponse>
+  }
+
+  async getRichList(req: Insights.QueryGetRichListRequest): Promise<Insights.InsightsQueryResponse<Insights.QueryGetRichListResponse>> {
```

Comment:
we should follow the naming structure of the new API, and forget the old. we can annotate the legacy names if necessary